src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt

```diff
@@ -0,0 +1,104 @@
+package pl.edu.agh.gem.internal.service
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import org.mockito.kotlin.any
+import org.mockito.kotlin.atLeastOnce
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import pl.edu.agh.gem.external.detector.AttachmentContentTypeNotSupportedException
+import pl.edu.agh.gem.external.detector.AttachmentSizeExceededException
+import pl.edu.agh.gem.helper.group.DummyGroup.GROUP_ID
+import pl.edu.agh.gem.helper.group.createGroupMembers
+import pl.edu.agh.gem.internal.client.GroupManagerClient
+import pl.edu.agh.gem.internal.detector.FileDetector
+import pl.edu.agh.gem.internal.persistence.GroupAttachmentRepository
+import pl.edu.agh.gem.util.createGroupAttachment
+
+class GroupServiceTest : ShouldSpec({
+    val groupAttachmentRepository = mock<GroupAttachmentRepository>()
+    val groupManagerClient = mock<GroupManagerClient> { }
+    val fileDetector = mock<FileDetector> { }
+    val groupService = GroupService(
+        groupAttachmentRepository = groupAttachmentRepository,
+        groupManagerClient = groupManagerClient,
+        fileDetector = fileDetector,
+    )
+
+    should("save group attachment") {
+        // given
+        val attachment = createGroupAttachment()
+        val data = attachment.file.data
+
+        whenever(fileDetector.getFileSize(data)).thenReturn(attachment.sizeInBytes)
+        whenever(fileDetector.getFileMediaType(data)).thenReturn(attachment.contentType)
+        whenever(groupAttachmentRepository.save(any())).thenReturn(attachment)
+
+        // when
+        val groupAttachment = groupService.saveAttachment(data, attachment.groupId, attachment.uploadedByUser)
+
+        // then
+        verify(groupAttachmentRepository, times(1)).save(any())
+        groupAttachment.also {
```

Comment:
Can't we just write it like that: ``` groupAttachment shouldBe attachment``` ?

Reply:
for me its easy to modify one place that 12

Reply:
changed

Reply:
you didn't

Reply:
true changed i an other place also